### PR TITLE
Make tests faster by default

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by keszybz** [comment]
Ah, I forgot: this only adds the meson part, so it should not be merged before #6266.

**Comment by martinpitt** [comment]
Semaphore fails:
```
src/test/test-watchdog.c:36:29: error: ‘SYSTEMD_SLOW_TESTS_DEFAULT’ undeclared (first use in this function)
         slow = r >= 0 ? r : SYSTEMD_SLOW_TESTS_DEFAULT;
                             ^
```

**Comment by keszybz** [comment]
There's no automake support.

**Comment by poettering** [comment]
Could you rebase this, now that #6266 has been merged?

**Comment by keszybz** [comment]
Is there a need to rebase? github says it can be merged w/o issue.

**Comment by fsateler** [comment]
CI should be configured to pass `-Dslow-tests=true` before merging, no?

**Comment by poettering** [comment]
Oh, no real need, just wanted to see the CI pass, if this is applied to current git, because the CIs so far didn't test that

**Comment by keszybz** [comment]
I added -Dslow-tests=true to thread#1 and -Db_sanitize=address to thread#2. Let's see.

**Comment by keszybz** [comment]
I deleted thread#2 and rebuilt again. Now it passes.

(We probably want to build with asan, but apparently this requires some additional deps.)

File: meson_options.txt [hunk 1]
```diff
@@ -252,5 +252,7 @@ option('zshcompletiondir', type : 'string',
 
 option('tests', type : 'combo', choices : ['true', 'unsafe'],
        description : 'enable extra tests with =unsafe')
+option('slow-tests', type : 'boolean', value : 'false',
+       description : 'run the slow tests by default')
```
Comment (martinpitt) [inline meson_options.txt:256]: This is bikeshedding matter of course, but shouldn't we keep the slow tests on by default, and only provide that option to turn them off? (i. e. `value: 'true'`)

File: src/resolve/test-resolved-packet.c [hunk 2]
```diff
@@ -31,6 +31,9 @@ static void test_dns_packet_new(void) {
 
                 log_debug("dns_packet_new: %zu → %zu", i, p->allocated);
                 assert_se(p->allocated >= MIN(DNS_PACKET_SIZE_MAX, i));
+
+                if (i > DNS_PACKET_SIZE_START + 10 && i < DNS_PACKET_SIZE_MAX - 10)
+                        i = MIN(i * 2, DNS_PACKET_SIZE_MAX - 10);
```
Comment (martinpitt) [inline src/resolve/test-resolved-packet.c:36]: Heh, nice exponential go-faster stripes 👍

File: meson_options.txt (hunk 1 above)
Comment (keszybz) [reply meson_options.txt:256]: I don't think that those "slow" tests bring that much value ;) It's good that we can hammer the hashmap code, but I think it'll be enough if we just do that on some fraction of builds — if it were to find any issues, that'll be enough. Similarly for the other "slow" tests. And it's good to make the common case fast, just to encourage users and developers to run the test suite often.

Comment (martinpitt) [reply meson_options.txt:256]: OK. How about we configure one of the semaphore tests to use slow tests, so that we at least have one code path with them?

Comment (martinpitt) [reply meson_options.txt:256]: Nevermind, I just saw that semaphore already builds with `-Dslow-tests=true`.

**Pull request merged**